Source/Provers/SMTLib/SMTLibLineariser.cs

```diff
@@ -459,7 +459,7 @@ public bool Visit(VCExprQuantifier node, LineariserOptions options)
         if (hasAttrs)
         {
           wr.Write("\n");
-          if (info.qid != null)
+          if (info.qid != null && CommandLineOptions.Clo.EmitSkolemIdAndQId)
```

Comment:
It looks like it is possible for info.qid to be null in which case nothing will be emitted.  So, there already appears to be a way to control whether something is emitted.  Then what is the purpose of the new option?

Reply:
> So, there already appears to be a way to control whether something is emitted.

Judging by the code's, there's not. There's only a few edge cases in which it's null but you can't trigger it to be null everywhere.

Reply:
I think `info.qid` is set only if the Boogie program uses the `{:qid ...}` attribute.

Reply:
You can see [here](https://github.com/boogie-org/boogie/blob/master/Source/VCExpr/Boogie2VCExpr.cs#L690) that Boogie generates a `{:qid}` if none is specified:

```
      if (qid == null && vars.Count != 0)
      {
        // generate default name (line:column position in .bpl file)
```

Also Dafny does not generate any `{:qid}` attributes but the resolving SMT logs still contain them.